### PR TITLE
Tsylia/1066 Unique Edrpou Ipn

### PR DESCRIPTION
**Pull request by andriitsylia**

### CODE REVIEW

File: OutOfSchool/OutOfSchool.WebApi/Services/ProviderService.cs [hunk 1]
```diff
@@ -484,6 +484,11 @@ private protected async Task<ProviderDto> UpdateProviderWithActionBeforeSavingCh
             throw new ArgumentNullException(nameof(userId));
         }
 
+        if (await ExistsAnotherProviderWithTheSameEdrpouIpn(providerUpdateDto))
+        {
+            return null;
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.WebApi/Services/ProviderService.cs:489]: Shouldn't any trace log be added here?

**Review by VadymLevkovskyi — APPROVED** [review]
Approved, though have minor question about handling errors. It looks like we are ignoring error reason that could be potentially harmful for prod cases debugging. 

I'm not sure what is OoS policy regarding this, so, @andriitsylia better check it with @DmyMi

File: OutOfSchool/OutOfSchool.WebApi/Services/ProviderService.cs (hunk 1 above)
Comment (andriitsylia) [reply OutOfSchool/OutOfSchool.WebApi/Services/ProviderService.cs:489]: Ok, I agree with you. I'll add logger.LogInformatiom() to this snippet of the code.

**Comment by DmyMi** [comment]
@VadymLevkovskyi yes the policy is to at least log the exception

**Review by VadymLevkovskyi — APPROVED** [review]

File: OutOfSchool/OutOfSchool.WebApi/Services/ProviderService.cs [hunk 2]
```diff
@@ -777,4 +784,14 @@ private async Task UpdateWorkshopsProviderStatus(Guid providerId, ProviderStatus
...
+    private async Task<bool> ExistsAnotherProviderWithTheSameEdrpouIpn(ProviderUpdateDto providerUpdateDto)
+    {
+        var providersWithTheSameEdrpouIpn = await providerRepository
+            .GetByFilter(x => x.EdrpouIpn == providerUpdateDto.EdrpouIpn)
+            .ConfigureAwait(false);
+
+        return providersWithTheSameEdrpouIpn.Any()
+            && providersWithTheSameEdrpouIpn.First().Id != providerUpdateDto.Id;
+    }
 }
```
[the remark this answers is not in the document]
Comment (andriitsylia) [reply OutOfSchool/OutOfSchool.WebApi/Services/ProviderService.cs:796]: I did it.

**Pull request merged**